modules/imgproc/src/stackblur.cpp

```diff
@@ -547,7 +547,7 @@ class ParallelStackBlurRow : public ParallelLoopBody
         }
         else
         {
-            size_t bufSize = CN * (width + radius) * sizeof(TBuf) + 2 * CN * sizeof(TBuf);
+            size_t bufSize = CN * (width + kernelSize) * sizeof(TBuf) + 2 * CN * sizeof(TBuf);
```

Comment:
`kernelSize = radius*2 +1`, if the kernel size is bigger then width, we need to set a bigger buffer for it.